src/middleware/session_auth.rs

```diff
@@ -28,29 +34,34 @@ impl<'r> FromRequest<'r> for SessionAuth {
         // TODO: use fairing for db connection?
         // let db = try_outcome!(request.guard::<Database>().await);
 
-        let mut db = match request.rocket().state::<Database>() {
-            Some(db) => db.conn(),
+        let db = match request.rocket().state::<Database>() {
+            Some(db) => db,
             None => {
-                return Outcome::Error((
-                    Status::InternalServerError,
-                    SessionAuthError::DatabaseConnection,
-                ))
+                return Outcome::Error((Status::Unauthorized, SessionAuthError::DatabaseConnection))
             }
         };
```

Comment:
nit:
```suggestion
        let db = request.rocket().state::<Database>()
            .ok_or_else(|| Outcome::Error((Status::Unauthorized, SessionAuthError::DatabaseConnection)))?;
```

Reply:
Unfortunately `Outcome` doesn't seem to work with the `?` operator